.github/workflows/ci_checks.yml

```diff
@@ -0,0 +1,18 @@
+name: General checks
+
+on:
+  push:
+    branches:
+      - "*"
+      - "**"
+      - "!publication"
```

Comment:
Nice! Checked that the order of these patterns matters and this is correct :) https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#example-including-and-excluding-branches-and-tags